features/support/webmock.rb

```diff
@@ -5,7 +5,8 @@
 
 require_relative 'capybara_driver_helper'
 
-selenium_url = URI.parse ENV.fetch('SELENIUM_URL', 'http://localhost:4444/wd/hub')
-app_host_url = URI.parse Capybara.app_host
+URI.parse ENV.fetch('SELENIUM_URL', 'http://localhost:4444/wd/hub')
+URI.parse Capybara.app_host
 
-WebMock.disable_net_connect!(allow_localhost: true, allow: [selenium_url.host, app_host_url.host, 'https://messages.cucumber.io/api/reports'])
+# WebMock.disable_net_connect!(allow_localhost: true, allow: [selenium_url.host, app_host_url.host, 'https://messages.cucumber.io/api/reports'])
+WebMock.allow_net_connect!
```

Comment:
Could you please tell more about this? We have disable net connect here so there are no calls to the outside of the test framework. And when there is such an attempt then the test will fail.

Reply:
smoke test failing on the following build 
https://build.platform.hmcts.net/job/HMCTS_d_to_i/job/hwf-publicapp/job/PR-670/8/console
I was getting an error related to api call .. so, I enabled WebMock.allow_net_connect!

Reply:
Did you take a look why the API was trigger? 

I run into the same problem on my PR for the pages swap. The problem here is that the method that mocks the API call was not called. Look at my PR https://github.com/hmcts/hwf-publicapp/pull/678/files specifically line 4 at features/e2e.feature.

I am happy to answer any questions about the test setup and tools we use and why.